lua/entities/gmod_wire_coldetector.lua

```diff
@@ -0,0 +1,247 @@
+AddCSLuaFile()
+
+if !WireLib then return end
+DEFINE_BASECLASS( "base_wire_entity" )
+ENT.PrintName       = "Wire Collision Detector"
+ENT.WireDebugName	= "Collision Detector"
+local f =""
+local i = 1
+/*if CLIENT then
+	while(true) do
+		name, value = debug.getupvalue(effects.Create,i)
+		if name == nil then break end
+		if name == "EffectList" then
+			PrintTable(value)
+		end
+		i=i+1
+	end
+end*/
+
+if CLIENT then return end
+local stickplayer=CreateConVar( "wire_col_detector_stickplayer", 0, 0, "Allow linked entity stick to players" )
+local stickprops=CreateConVar( "wire_col_detector_stickprops", 0, 0, "Allow sticking to other players stuff" )
+function entitiesCollide(ent,data)
+	if !IsValid(ent.ColSensor) || ent.ColSensor.LinkedEnt != ent then return end
+	ent.ColSensor:ApplyCollide(data,ent)
+end
+
+function ENT:Initialize()
+	self:PhysicsInit( SOLID_VPHYSICS )
+	self:SetMoveType( MOVETYPE_VPHYSICS )
+	self:SetSolid( SOLID_VPHYSICS )
+	self.LinkedEnt=self
+	self.Inputs = WireLib.CreateInputs( self, { "Stick to world", "Stick to props", "Stick to npc/player"} )
+	self.Outputs = WireLib.CreateOutputs(self, {"Collided","Speed","Delta Time","Hit Position [VECTOR]","Hit Normal [VECTOR]","Velocity [VECTOR]","Hit Entity Velocity [VECTOR]","Hit Entity [ENTITY]","Entity Position [VECTOR]","Entity Angle [ANGLE]","Hit Entity Position [VECTOR]","Hit Entity Angle [ANGLE]"})
+	//self:Setup(1,1,false,false,0,0,0,0,0,0,false,"","","",0,0,0,false,false,0,0,false,false,false,false)
+end
+function ENT:Setup(AllowConstrained,NoPhysics)
+	self.AllowConstrained=AllowConstrained
+	self.NoPhysics=NoPhysics
+	self:ShowOutput()
+end
+
```

Comment:
Add more empty lines between functions, in order to set them apart visually.
If you want to group functions together, use two newlines around the groups